### PR TITLE
Implemented SA1500 diagnostic tests

### PR DESCRIPTION
**Pull request by vweijsters**
Also see issue #179

The failing tests are being skipped, in order to keep the AppVeyor build working.

I've split up the tests over a number of partial classes, in order to improve readability and navigation. Let me know how that works out for you guys.

These tests are inspired by the test code provided by StyleCop and should (hopefully) be quite rigorous.

There is probably some duplication in there that could be eliminated, but that is for the future imho.

The code is largely untested. I've used StyleCop to verify the test code as much as possible and to verify the position of the diagnostics where possible.

This one turned out to be much larger than I initially expected :sweat:

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1500UnitTests.Properties.cs [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    public bool Property10 { get { return this.test; } 
+    }
+
+    // Invalid property #11
+    public bool Property11
+    { get { return this.test; } 
+    }
+
+    // Invalid property #12
+    public int[] Property12 { get; set; } =
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1500UnitTests.Properties.cs]: I wouldnt have thought about that. Nice!

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1500UnitTests.StatementBlocks.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    using Xunit;
+    using System.Collections.Generic;
+
+
+    /// <summary>
+    /// Unit tests for <see cref="SA1500CurlyBracketsForMultiLineStatementsMustNotShareLine"/>.
+    /// </summary>
+    public partial class SA1500UnitTests : DiagnosticVerifier
+    {
+        public static IEnumerable<object[]> StatementBlocksTokenList
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1500UnitTests.StatementBlocks.cs]: Couldnt you include fixed and foreach in here and remove the special tests for them?

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/StyleCop.Analyzers.Test.csproj [hunk 3]
```diff
@@ -136,6 +138,29 @@
     <Compile Include="Helpers\CodeFixVerifier.Helper.cs" />
     <Compile Include="Helpers\DiagnosticResult.cs" />
     <Compile Include="Helpers\DiagnosticVerifier.Helper.cs" />
+    <Compile Include="LayoutRules\SA1500UnitTests.Blocks.cs" />
```
Comment (pdelvo) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/StyleCop.Analyzers.Test.csproj]: Maybe it might be a good idea to put all of those into a new directory?

**Comment by pdelvo** [comment]
Wow thats huge! I looked over it and it looks very good. Thats way more then I would have done ;)
I think there are a few places where it is possible to reduce a bit of duplication, but I dont think that this is a big deal.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/StyleCop.Analyzers.Test.csproj (hunk 3 above)
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/StyleCop.Analyzers.Test.csproj]: That is a good idea, I'll do that.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1500UnitTests.StatementBlocks.cs (hunk 2 above)
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1500UnitTests.StatementBlocks.cs]: I had a look and that will work -> commit incoming :wink:

**Comment by sharwell** [comment]
Blocked on #2

**Comment by sharwell** [comment]
I merged this into #2.

**Pull request closed without merging**